### DISCUSSION TITLE
Are there any plans on bringing in cached features in the future?

### DISCUSSION THREAD

**Discussion by Raj2032** [emilk/egui#1115] [General]
Hey guys, I am wondering are there any plans on bringing in cached features in the future?

Or is there a way to cache certain things, for example:

![image](https://user-images.githubusercontent.com/29293830/149628465-ed63188f-59db-4697-bf77-c9ccd67af376.png)

See if I move the slider, everything has to update, even the `Click each year` button (where the red arrow is pointing to) which is a waste of CPU usage.

Would there be a way to cache the elements that don't need to be updated?

**Comment by emilk** [comment]
No, at least not on such a coarse level. This is what immediate mode is. If you want caching, use a retained mode library!

Perhaps we could add caching of separate panels and windows though. But right now I would rather spend that time optimizing egui.
